static/app/views/dashboards/detail.spec.tsx

```diff
@@ -2020,6 +2020,100 @@ describe('Dashboards > Detail', function () {
       );
     });
 
+    it('renders favorite button in unfavorited state', async function () {
+      MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/dashboards/1/',
+        body: DashboardFixture([], {id: '1', title: 'Custom Errors', isFavorited: false}),
+      });
+      render(
+        <ViewEditDashboard
+          {...RouteComponentPropsFixture()}
+          organization={initialData.organization}
+          params={{orgId: 'org-slug', dashboardId: '1'}}
+          router={initialData.router}
+          location={initialData.router.location}
+        >
+          {null}
+        </ViewEditDashboard>,
+        {
+          router: initialData.router,
+          organization: {
+            features: ['dashboards-favourite', ...initialData.organization.features],
+          },
+        }
+      );
+
+      const favoriteButton = await screen.findByLabelText('dashboards-favourite');
+      expect(favoriteButton).toBeInTheDocument();
+      expect(screen.getByTestId('empty-star')).toBeInTheDocument();
```

Comment:
See my other comment to change this so it's testing something more semantic with the label. We typically try to avoid test IDs as much as possible and aim for things that are more semantic

We try to keep these in mind: https://testing-library.com/docs/queries/about#priority